metricbeat/module/kibana/stats/stats.go

```diff
@@ -122,36 +121,39 @@ func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
 // Fetch methods implements the data gathering and data conversion to the right format
 // It returns the event which is then forward to the output. In case of an error, a
 // descriptive error must be returned.
-func (m *MetricSet) Fetch(r mb.ReporterV2) {
+func (m *MetricSet) Fetch(r mb.ReporterV2) error {
 	now := time.Now()
 
-	m.fetchStats(r, now)
+	err := m.fetchStats(r, now)
```

Comment:
Good catch.

Reply:
BTW, just realised this might be odd when x-pack is enabled and we send up an error event ...

Reply:
Yes, good catch. I'll have to revert this change.

Reply:
Actually, it won't be odd because of this line: https://github.com/elastic/beats/pull/11763/files#diff-8710596f0decaae3ca552cedea1eb0a0R150

Reply:
I added a comment around that line now so hopefully it's a bit clearer now.